array-reduce/main.js

```diff
@@ -0,0 +1,33 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const previousValue = 0;
+
+const sum = numbers.reduce((previousValue, currentValue) => previousValue + currentValue);
+console.log('sum:', sum);
+
+const product = numbers.reduce((previousValue, currentValue) => previousValue * currentValue);
+console.log('product', product);
+
+const balance = account.reduce((previousValue, currentValue) => previousValue + currentValue.amount, previousValue);
+console.log('balance:', balance);
+
+const obj = {};
+const composite = traits.reduce((previousValue, currentValue) => Object.assign(obj, currentValue), previousValue);
+console.log('composite:', composite);
```

Comment:
Try using `obj` as your initial value argument here, then pass `previousValue` and `currentValue` as your `Object.assign()` arguments.

Reply:
got it